### DISCUSSION TITLE
[Form|Fix] `Builder` Component `Block` Previews

### DISCUSSION THREAD

**Discussion by bezhanSalleh** [filamentphp/filament#14582] [Ideas]
The `Block` preview feature is a nice addition to the form `Builder` component. Adding a `preview` is specific to each block while enabling the previews for them is done in the parent component `Builder` by calling the `blockPreviews()` method. Now here is the thing , if you enable the `blockPreviews()` each of your `Block`s must have a preview otherwise it will throw an error of `view()` not found. But for some `Block`s previews make sense and for some it doesn't and sometimes you want this ability.

**Easy Fix** without any Breaking Changes
1. Introducing a new method in the following `trait` to check if the view for the block was provided or not
https://github.com/filamentphp/filament/blob/56a175a12909e8c76c423a50c90c1d507493d2e4/packages/forms/src/Components/Concerns/HasPreview.php#L1-L29
    ```php
        public function shouldPreview(): bool
        {
            return (bool) filled($this->evaluate($this->preview));
        }
    ```
2. Updating the relevant part with our new method 
https://github.com/filamentphp/filament/blob/56a175a12909e8c76c423a50c90c1d507493d2e4/packages/forms/resources/views/components/builder.blade.php#L231
    ```php
    'p-4' => ! ($hasBlockPreviews && $item->getParentComponent()->shouldPreview()),
    ```
https://github.com/filamentphp/filament/blob/56a175a12909e8c76c423a50c90c1d507493d2e4/packages/forms/resources/views/components/builder.blade.php#L234
    ```php
    @if ($hasBlockPreviews && $item->getParentComponent()->shouldPreview())
    ```
@danharrin heyo mate! let me know if it should be PR'ed or not. Cheers!

**Comment by danharrin** [comment]
Sounds good, please name it `hasPreview()` though

**Reply by bezhanSalleh** [reply]
> Sounds good, please name it `hasPreview()` though

Done! #14592
